src/App.js

```diff
@@ -1,16 +1,26 @@
-import React from "react";
+import React, { useEffect } from "react";
 import TodoList from "./TodoList";
 import AddTodoForm from "./AddTodoForm";
+ function useSemiPersistentState(){
+  const[todoList, setTodoList]= React.useState(JSON.parse(localStorage.getItem("savedTodoList"))||[]);
 
-function App() {
+  useEffect(()=>{
+    localStorage.setItem("savedTodoList",JSON.stringify(todoList));
+   },[todoList]
+  )
+return(
+  [todoList,setTodoList])
+ }
  
+function App() {
+const[todoList, setTodoList]=useSemiPersistentState();
+
+   
 
-    const[todoList, setTodoList]= React.useState([]);
     const addTodo =(newTodo)=>{
-      console.log(newTodo);
- setTodoList([...todoList,newTodo]);
+     setTodoList([...todoList,newTodo]);
       
-    }
+   }
   return (
     <div>
```

Comment:
The lesson instructions said to use a fragment in the JSX. It's a minor error, but check that you know what a fragment is and when to use it.